2020-10-06 10:53:04,234 : INFO : root : Loading config.
2020-10-06 10:53:04,257 : INFO : YamlMultifileConfigLoader : Machine config file #1: config.yaml
2020-10-06 10:53:04,259 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\2b6ef079982d662fef43bd5e7e2d4708.mpf_cache
2020-10-06 10:53:04,482 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\89b9c5cb952cce9fc74d723399719cc6.mpf_cache
2020-10-06 10:53:04,482 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\4fcecb70f81c3c2fefb8e65ae9a0af5f.mpf_cache
2020-10-06 10:53:04,498 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\5820a65357aa32f3082a694bbd0b1b93.mpf_cache
2020-10-06 10:53:04,498 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\2f1df67e0c6599f893aeb5468bc976f9.mpf_cache
2020-10-06 10:53:04,498 : WARNING : ConfigProcessor : Config file in cache changed: c:\GitHub\Cuphead\modes\bonus\config\bonus.yaml
2020-10-06 10:53:04,498 : INFO : ConfigProcessor : Loading config from file c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf\modes\bonus\config\bonus.yaml.
2020-10-06 10:53:04,498 : INFO : ConfigProcessor : Loading config: c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf\modes\bonus\config\bonus.yaml
2020-10-06 10:53:04,498 : INFO : ConfigProcessor : Loading config from file c:\GitHub\Cuphead\modes\bonus\config\bonus.yaml.
2020-10-06 10:53:04,529 : INFO : ConfigProcessor : Loading config: c:\GitHub\Cuphead\modes\bonus\config\bonus.yaml
2020-10-06 10:53:04,529 : INFO : ConfigProcessor : Config file cache created: C:\Users\themi\AppData\Local\Temp\3e28196fea866bd961d93515ab5c664b.mpf_cache
2020-10-06 10:53:04,529 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\893c92602cb79b5461df4ae912d665e0.mpf_cache
2020-10-06 10:53:04,529 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\15cb35af48ef2b73a1399ecc64f542cf.mpf_cache
2020-10-06 10:53:04,529 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\8f78dc3447c1c9fe6775dbe5408c12e9.mpf_cache
2020-10-06 10:53:04,529 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\459e0fbd95aa261661d4ec522fde7178.mpf_cache
2020-10-06 10:53:04,529 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\bc011c11c9e34bbd89eedfa08565ed62.mpf_cache
2020-10-06 10:53:04,529 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\6f009ec52b9ccf4b89f291e4051b3241.mpf_cache
2020-10-06 10:53:04,545 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\c2ce7139d0aaca9aa552320269b3f50d.mpf_cache
2020-10-06 10:53:04,545 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\2b7bf9166be80f4b554569a332b23de4.mpf_cache
2020-10-06 10:53:04,545 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\c546e3aedc6779039624162d0203f540.mpf_cache
2020-10-06 10:53:04,545 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\abdfe7fa7b37a4a193490f565dc0597d.mpf_cache
2020-10-06 10:53:04,545 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\4342b421d288ce5a4483067739b72cdd.mpf_cache
2020-10-06 10:53:04,545 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\368aa16bb3be9e41d99a527245060f83.mpf_cache
2020-10-06 10:53:04,545 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\d0e012eb2507157abb0bb72425aaf5b2.mpf_cache
2020-10-06 10:53:04,545 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\0f1b51b912b71211c13fe22b9a37b50a.mpf_cache
2020-10-06 10:53:04,545 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\10ee2574e78165e598e7a995071a74cc.mpf_cache
2020-10-06 10:53:04,545 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\a266cad92e011305e896e67dddfd4a6c.mpf_cache
2020-10-06 10:53:04,545 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\7ee3e0eba3dedef43a01da4ffa491384.mpf_cache
2020-10-06 10:53:04,545 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\13fd6d25ddf06daef684ee2be33938b7.mpf_cache
2020-10-06 10:53:04,560 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\0b0cb9dce530fd5533b52f679ba6d4c7.mpf_cache
2020-10-06 10:53:04,560 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\6448a7490ccbbf7c92ae3827fedf65d7.mpf_cache
2020-10-06 10:53:04,560 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\cbfe20b556c038913af5118889400ccc.mpf_cache
2020-10-06 10:53:04,560 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\db00c3dd6414ff9cfb2bf542763a8a69.mpf_cache
2020-10-06 10:53:04,560 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\1fa5e27674b4aa7d98e43b722406aa74.mpf_cache
2020-10-06 10:53:04,560 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\5883b660032b5311c8838643c5c4f04e.mpf_cache
2020-10-06 10:53:04,560 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\49716d4cb57f232d2d55ea4b85c0bf9a.mpf_cache
2020-10-06 10:53:04,560 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\02d9d2d34f4e72c61edea96a1832348d.mpf_cache
2020-10-06 10:53:04,560 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\75ddffbe1d195e10645833a8c787820e.mpf_cache
2020-10-06 10:53:04,560 : INFO : ConfigProcessor : Loading config from cache: C:\Users\themi\AppData\Local\Temp\383ac4b0678bf6b0edc3780467155495.mpf_cache
2020-10-06 10:53:04,560 : INFO : Machine : Mission Pinball Framework Core Engine v0.54.0-dev.72
2020-10-06 10:53:04,560 : INFO : Machine : Command line arguments: {'no_load_cache': False, 'create_config_cache': True, 'bcp': True, 'configfile': ['config.yaml'], 'force_assets_load': False, 'jsonlogging': False, 'logfile': 'logs\\2020-10-06-10-53-04-mpf-LAPTOP-C39PEUFD.log', 'pause': False, 'production': False, 'text_ui': True, 'loglevel': 15, 'consoleloglevel': 20, 'force_platform': 'smart_virtual', 'syslog_address': None, 'mc_file_name': None, 'no_sound': False}
2020-10-06 10:53:04,560 : INFO : Machine : MPF path: c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf
2020-10-06 10:53:04,560 : INFO : Machine : Machine path: c:\GitHub\Cuphead
2020-10-06 10:53:04,560 : INFO : Machine : Platform: win32
2020-10-06 10:53:04,560 : INFO : Machine : Python executable location: c:\users\themi\appdata\local\programs\python\python36\python.exe
2020-10-06 10:53:04,560 : INFO : Machine : Python version: 3.6.4 (64-bit)
2020-10-06 10:53:04,576 : INFO : Machine : Initialise MPF.
2020-10-06 10:53:05,060 : INFO : EventManager : Event: ======'machine_var_credits_string'====== Args={'value': 'FREE PLAY', 'prev_value': None, 'change': True}
2020-10-06 10:53:05,201 : INFO : EventManager : Event: ======'machine_var_player1_score'====== Args={'value': 16340, 'prev_value': None, 'change': True}
2020-10-06 10:53:05,201 : INFO : EventManager : Event: ======'machine_var_mpf_version'====== Args={'value': 'MPF v0.54.0-dev.72', 'prev_value': None, 'change': True}
2020-10-06 10:53:05,217 : INFO : EventManager : Event: ======'machine_var_mpf_extended_version'====== Args={'value': 'MPF v0.54.0-dev.72, Config version:5, Show version: 5, BCP version:1.1', 'prev_value': None, 'change': True}
2020-10-06 10:53:05,217 : INFO : EventManager : Event: ======'machine_var_python_version'====== Args={'value': '3.6.4', 'prev_value': None, 'change': True}
2020-10-06 10:53:05,217 : INFO : EventManager : Event: ======'machine_var_platform'====== Args={'value': 'Windows-10-10.0.18362-SP0', 'prev_value': None, 'change': True}
2020-10-06 10:53:05,217 : INFO : EventManager : Event: ======'machine_var_platform_system'====== Args={'value': 'Windows', 'prev_value': None, 'change': True}
2020-10-06 10:53:05,217 : INFO : EventManager : Event: ======'machine_var_platform_release'====== Args={'value': '10', 'prev_value': None, 'change': True}
2020-10-06 10:53:05,217 : INFO : EventManager : Event: ======'machine_var_platform_version'====== Args={'value': '10.0.18362', 'prev_value': None, 'change': True}
2020-10-06 10:53:05,217 : INFO : EventManager : Event: ======'machine_var_platform_machine'====== Args={'value': 'AMD64', 'prev_value': None, 'change': True}
2020-10-06 10:53:05,217 : INFO : EventManager : Event: ======'init_phase_1'====== Args={}
2020-10-06 10:53:05,373 : INFO : EventManager : Event: ======'drop_target_bank_cup_bank_up'====== Args={}
2020-10-06 10:53:05,373 : INFO : EventManager : Event: ======'drop_target_bank_mug_bank_up'====== Args={}
2020-10-06 10:53:05,389 : INFO : EventManager : Event: ======'init_phase_2'====== Args={}
2020-10-06 10:53:05,389 : INFO : SwitchController : Found active switch: ('2', <Platform.SmartVirtual>)
2020-10-06 10:53:05,389 : INFO : SwitchController : Found active switch: ('3', <Platform.SmartVirtual>)
2020-10-06 10:53:05,389 : ERROR : asyncio : Exception in callback EventManager._queue_task_done(<Task finishe...s \'int\'>',)>)
handle: <Handle EventManager._queue_task_done(<Task finishe...s \'int\'>',)>)>
Traceback (most recent call last):
  File "c:\users\themi\appdata\local\programs\python\python36\lib\asyncio\events.py", line 145, in _run
    self._callback(*self._args)
  File "c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf\core\events.py", line 716, in _queue_task_done
    future.result()
  File "c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf\core\events.py", line 633, in _run_handlers_sequential
    handler.callback(queue=queue, **merged_kwargs)
  File "c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf\core\mode_controller.py", line 104, in initialise_modes
    **item.kwargs)
  File "c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf\core\config_player.py", line 133, in process_mode_config
    config = self.validate_config(config)
  File "c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf\core\config_player.py", line 105, in validate_config
    validated_config[event] = self.validate_config_entry(settings, event)
  File "c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf\config_players\flat_config_player.py", line 17, in validate_config_entry
    config = self._parse_config(settings, name)
  File "c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf\core\config_player.py", line 127, in _parse_config
    return self.get_full_config(config)
  File "c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf\config_players\flat_config_player.py", line 23, in get_full_config
    value[element] = super().get_full_config(value[element])
  File "c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf\core\config_player.py", line 199, in get_full_config
    self.config_file_section, value, base_spec='config_player_common')
  File "c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf\core\config_validator.py", line 176, in validate_config
    validation_failure_info=validation_failure_info)
  File "c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf\core\config_validator.py", line 194, in _validate_config
    source.__class__
  File "c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf\core\config_validator.py", line 718, in validation_error
    item, msg), 5 if code is None else code, self.log.name)
mpf.exceptions.config_file_error.ConfigFileError: Config File Error in ConfigValidator: Config validation error: Entry event_player:event_player = "-1" is not valid. Config attribute should be dict but is <class 'int'> Error Code: CFE-ConfigValidator-5 (https://docs.missionpinball.org/en/dev/logs/CFE-ConfigValidator-5.html)
2020-10-06 10:53:05,498 : INFO : Machine : Starting the main run loop.
2020-10-06 10:53:05,514 : INFO : Machine : Shutting down...
2020-10-06 10:53:05,514 : INFO : EventManager : Event: ======'shutdown'====== Args={}
2020-10-06 10:53:05,514 : ERROR : Machine : Runtime Exception
Traceback (most recent call last):
  File "c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf\core\machine.py", line 745, in _run_loop
    raise self._exception['exception']
  File "c:\users\themi\appdata\local\programs\python\python36\lib\asyncio\events.py", line 145, in _run
    self._callback(*self._args)
  File "c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf\core\events.py", line 716, in _queue_task_done
    future.result()
  File "c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf\core\events.py", line 633, in _run_handlers_sequential
    handler.callback(queue=queue, **merged_kwargs)
  File "c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf\core\mode_controller.py", line 104, in initialise_modes
    **item.kwargs)
  File "c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf\core\config_player.py", line 133, in process_mode_config
    config = self.validate_config(config)
  File "c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf\core\config_player.py", line 105, in validate_config
    validated_config[event] = self.validate_config_entry(settings, event)
  File "c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf\config_players\flat_config_player.py", line 17, in validate_config_entry
    config = self._parse_config(settings, name)
  File "c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf\core\config_player.py", line 127, in _parse_config
    return self.get_full_config(config)
  File "c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf\config_players\flat_config_player.py", line 23, in get_full_config
    value[element] = super().get_full_config(value[element])
  File "c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf\core\config_player.py", line 199, in get_full_config
    self.config_file_section, value, base_spec='config_player_common')
  File "c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf\core\config_validator.py", line 176, in validate_config
    validation_failure_info=validation_failure_info)
  File "c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf\core\config_validator.py", line 194, in _validate_config
    source.__class__
  File "c:\users\themi\appdata\local\programs\python\python36\lib\site-packages\mpf\core\config_validator.py", line 718, in validation_error
    item, msg), 5 if code is None else code, self.log.name)
mpf.exceptions.config_file_error.ConfigFileError: Config File Error in ConfigValidator: Config validation error: Entry event_player:event_player = "-1" is not valid. Config attribute should be dict but is <class 'int'> Error Code: CFE-ConfigValidator-5 (https://docs.missionpinball.org/en/dev/logs/CFE-ConfigValidator-5.html)
2020-10-06 10:53:05,529 : INFO : root : MPF run loop ended.
